pkg/streamingpromql/types/data_test.go

```diff
@@ -0,0 +1,122 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package types
+
+import (
+	"testing"
+
+	"github.com/prometheus/prometheus/model/histogram"
+	"github.com/prometheus/prometheus/promql"
+	"github.com/stretchr/testify/require"
+)
+
+func TestInstantVectorSeriesDataIterator(t *testing.T) {
+	type expected struct {
+		T       int64
+		F       float64
+		H       *histogram.FloatHistogram
+		HasNext bool
+	}
+	type testCase struct {
+		name     string
+		data     InstantVectorSeriesData
+		expected []expected
+	}
+
+	testCases := []testCase{
+		{
+			name: "floats only",
+			data: InstantVectorSeriesData{
+				Floats: []promql.FPoint{
+					{T: 1000, F: 1.1},
+					{T: 2000, F: 2.2},
+					{T: 3000, F: 3.3},
+				},
+			},
+			expected: []expected{
+				{1000, 1.1, nil, true},
+				{2000, 2.2, nil, true},
+				{3000, 3.3, nil, true},
+				{0, 0, nil, false},
+			},
+		},
+		{
+			name: "histograms only",
+			data: InstantVectorSeriesData{
+				Histograms: []promql.HPoint{
+					{T: 1500, H: &histogram.FloatHistogram{Sum: 1500}},
+					{T: 2500, H: &histogram.FloatHistogram{Sum: 2500}},
+				},
+			},
+			expected: []expected{
+				{1500, 0, &histogram.FloatHistogram{Sum: 1500}, true},
+				{2500, 0, &histogram.FloatHistogram{Sum: 2500}, true},
+				{0, 0, nil, false},
+			},
+		},
+		{
+			name: "mixed data",
```

Comment:
Would be good to add a test for the case where there is mixed data and the last point is a float (to contrast with this test case, where the last point is a histogram).